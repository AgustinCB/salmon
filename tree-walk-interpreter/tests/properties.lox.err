There was an error! [line 20] Error: Only instances have properties
