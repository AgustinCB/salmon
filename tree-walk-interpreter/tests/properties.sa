class DevonshireCream {
  serveOn() {
    return "Scones";
  }
  echo(a) {
    return a;
  }
}

var c = DevonshireCream();
c.test = 1;
print c.test;
c.testFunction = fun () {
    print "Scones!";
};
c.testFunction();
print c.serveOn();
print c.echo("hola");
var a = 1;
a.test = 1;
