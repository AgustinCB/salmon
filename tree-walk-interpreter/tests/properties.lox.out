1
Scones!
Scones
hola
